lua/nvim-tree/explorer/watch.lua

```diff
@@ -22,6 +24,29 @@ local function is_git(path)
   return path:match "%.git$" ~= nil or path:match(utils.path_add_trailing ".git") ~= nil
 end
 
+local function refresh_path(path)
+  local n = utils.get_node_from_path(path)
+  if not n then
+    return
+  end
+  log.line("watcher", "node event '%s'", path)
+
+  if M.refreshing_paths[path] then
+    log.line("watcher", "node event already running")
+    return
+  end
+  M.refreshing_paths[path] = true
```

Comment:
this is already managed by the debounce function right ? as this is synchronous it should not run 2 events in parallel anyway.

Reply:
Oh bugger... that should not have been committed. It was from "2. Queue one refresh when there is one in progress".

Removing, thanks for spotting.